asciimatics/widgets/textbox.py

```diff
@@ -114,16 +116,32 @@ def reset(self):
         # Reset to original data and move to end of the text.
         self._start_line = 0
         self._start_column = 0
-        self._line = len(self._value) - 1
+        if self._auto_scroll or self._line > len(self._value) - 1:
+            self._line = len(self._value) - 1
+
         self._column = 0 if self._is_disabled else len(self._value[self._line])
         self._reflowed_text_cache = None
 
+    def _draw_cursor(self, char, frame_no, x, y):
+        if not self._hide_cursor:
```

Comment:
Since this is only called in one place (that already checks whether it should call the function), I think it is less surprising if this check is added to that location rather than here.